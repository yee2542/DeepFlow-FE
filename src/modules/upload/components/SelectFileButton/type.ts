type SelectFileButtonPropsType = {
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
};

export type { SelectFileButtonPropsType };
