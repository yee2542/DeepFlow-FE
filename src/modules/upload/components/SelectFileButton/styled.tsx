import styled from 'styled-components';

export const UploadBtnWrapper = styled.div`
  input[type='file'] {
    width: 0.1px;
    height: 0.1px;
    opacity: 0;
    overflow: hidden;
    position: absolute;
    z-index: -1;
  }

  input[type='file'] + label {
    font-size: 1.25em;
    font-weight: 700;
    color: #5a36cc;
    background-color: #f8f5ff;
    border: 1px solid #d5c8ff;
    display: inline-block;
    cursor: pointer;
    border-radius: 4px;
    padding: 30px 40px 30px 40px;
    text-align: center;
  }

  input[type='file'] + label:hover {
    background-color: #ebe4ff;
  }
`;
