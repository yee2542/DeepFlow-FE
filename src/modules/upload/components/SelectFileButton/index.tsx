import React from 'react';
import { Wrapper } from './styled';
import { PictureOutlined } from '@ant-design/icons';
import { SelectFileButtonPropsType } from './type';

const SelectFileButton: React.FC<SelectFileButtonPropsType> = (props) => {
  const { onChange } = props;
  return (
    <Wrapper>
      <input type="file" name="file" id="file" multiple onChange={onChange} />
      <label htmlFor="file">
        {/* //TODO: wtf &nbsp */}
        + &nbsp; &nbsp; &nbsp;
        <PictureOutlined /> &nbsp; &nbsp; Select Photos
      </label>
    </Wrapper>
  );
};

export default SelectFileButton;
