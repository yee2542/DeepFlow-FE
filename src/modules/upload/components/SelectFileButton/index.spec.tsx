import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import React from 'react';
import SelectFileButton from '.';

describe('Component/SelectFileButton test', () => {
  it('Should match snapshot', () => {
    const wrap = renderer
      .create(
        <SelectFileButton
          onChange={() => {
            console.log('test');
          }}
        />,
      )
      .toJSON();
    expect(wrap).toMatchSnapshot();
  });

  it('SelectFileButton should call onChange with files when input change', () => {
    const onChange = jest.fn();
    const mockFile = new File(['hello'], 'hello.png', { type: 'image/png' });
    const expectResult = {
      target: {
        files: [mockFile],
      },
    };

    const wrapButton = shallow(<SelectFileButton onChange={onChange} />);
    wrapButton.find('input').simulate('change', { target: { files: [mockFile] } });
    expect(wrapButton.exists('input')).toBe(true);
    expect(onChange).toHaveBeenCalledWith(expectResult);
    expect(onChange).toHaveBeenCalledTimes(1);
  });
});
