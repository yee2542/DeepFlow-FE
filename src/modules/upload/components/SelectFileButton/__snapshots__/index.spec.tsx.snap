// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/SelectFileButton test Should match snapshot 1`] = `
<div
  className="styled__Wrapper-sc-23o88-0 dWCXYo"
>
  <input
    id="file"
    multiple={true}
    name="file"
    onChange={[Function]}
    type="file"
  />
  <label
    htmlFor="file"
  >
    +
    <span
      className="styled__IconWrapper-sc-23o88-1 ihOaOX"
    >
      <span
        aria-label="picture"
        className="anticon anticon-picture"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="picture"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 632H136v-39.9l138.5-164.3 150.1 178L658.1 489 888 761.6V792zm0-129.8L664.2 396.8c-3.2-3.8-9-3.8-12.2 0L424.6 666.4l-144-170.7c-3.2-3.8-9-3.8-12.2 0L136 652.7V232h752v430.2zM304 456a88 88 0 100-176 88 88 0 000 176zm0-116c15.5 0 28 12.5 28 28s-12.5 28-28 28-28-12.5-28-28 12.5-28 28-28z"
          />
        </svg>
      </span>
    </span>
    Select Photos
  </label>
</div>
`;
